File: 525.contiguous-array.cpp
/*
 * @lc app=leetcode id=525 lang=cpp
 *
 * [525] Contiguous Array
 */

// @lc code=start
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        int c0=0,c1=0,res=0,max1=0;
        for(int p=0;p<nums.size();p++)
        {
            c0=0;
            c1=0;
        for(int q=p;q<nums.size();q++)
        {
            int i=nums[q];
            i==0?c0++:c1++;
            if(c0==c1)res=c0+c1;
            max1 = max(res,max1);
        }
        }
        return max1;
    }
};
// @lc code=end

File: 525-contiguous-array/525-contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        if len(nums)<2:
            return 0
        res = {}
        tot = 0
        nums1 = nums[:]
        for i in range(len(nums1)):
            if nums1[i]==0:
                nums1[i]=-1
            if i==0:
                res[nums1[0]]=[0,0]
            if i>0:
                nums1[i]+=nums1[i-1]
                if nums1[i]==0:
                    tot = i+1
                if nums1[i] not in res:
                    res[nums1[i]]=[i,i]
                else:
                    res[nums1[i]][1]=i
        for i in res:
            tot = max(tot,res[i][1]-res[i][0])
        if tot%2!=0:
            tot+=1
        return tot
            